roxie/ccd/ccdactivities.cpp

```diff
@@ -2661,28 +2661,52 @@ class CRoxieIndexActivity : public CRoxieKeyedActivity
     bool sendContinuation(IMessagePacker * output)
     {
         MemoryBuffer si;
-        unsigned short siLen = 0;
+        unsigned siLen = 0;
         si.append(siLen);
+        si.append(false);
         si.append(lastRowCompleteMatch);
         si.append(inputsDone);
         si.append(processed);
         si.append(keyprocessed);
         si.append(lastPartNo.partNo);
         si.append(lastPartNo.fileNo);
         tlk->serializeCursorPos(si);
-        if (si.length() <= maxContinuationSize)
+        if (si.length() <= continuationCompressThreshold)
         {
             siLen = si.length() - sizeof(siLen);
             si.writeDirect(0, sizeof(siLen), &siLen);
             output->sendMetaInfo(si.toByteArray(), si.length());
             return true;
         }
         else
-            return false;
+        {
+            MemoryBuffer compressed;
+            compressed.append(siLen);  // Leaving space to patch when size known
+            compressed.append(true);
+            compressToBuffer(compressed, si.length() - compressed.length(), si.toByteArray() + compressed.length());
+            bool report = logctx.queryTraceLevel() && (logctx.queryTraceLevel() > 3 || si.length() <= continuationWarnThreshold);
```

Comment:
Shouldn't that be >= continuationWarnThreshold?

Reply:
Probably - added that bit of code while cleaning up the tracing before pushing.... far too simple to need testing!